doc/man3/PKCS12_newpass.pod

```diff
@@ -17,6 +17,16 @@ PKCS12_newpass() changes the password of a PKCS12 structure.
 B<p12> is a pointer to a PKCS12 structure. B<oldpass> is the existing password
 and B<newpass> is the new password.
 
+=head1 NOTES
+
+Each of B<oldpass> and B<newpass> is independently interpreted as a string in
+the UTF-8 encoding. If it is not valid UTF-8, it is assumed to be ISO9859-1
+instead.
+
+In particular, this means that passwords obtained through the default
+UI_OpenSSL() method, which are in the locale character set, should not be
+used without potentially being converted to UTF-8 first.
+
```

Comment:
The first says that it'll be interpreted as ISO8859-1 *if* it can't be interpreted as UTF-8. Which means that if it's some arbitrary string in ISO8859-1... then you need to do what `apps/pkcs11.c` does and convert it to UTF-8. You can't just pass it through blindly and pray that it isn't one of the corner cases that gets misinterpreted.